### DISCUSSION TITLE
Docker compose working directory?

### DISCUSSION THREAD

**Discussion by anupsathya** [firefly-iii/firefly-iii#5649] [Firefly III]
This might be a really dumb question but I'm trying to set up letsencrypt(certbot) and I can't seem to be able to find the working directory for firefly. I ran a server wide find for www-data and nothing showed up. This on a VPS running Debian 10. I used the default docker-compose.yml without any changes. 

Other information:
/var/www doesn't exist
The directory where I put docker-compose.yml doesn't work. 

Did I mess something up during the installation? Or am I missing something very simple?

**Comment by JC5** [comment]
If you bash into the container and do `pwd` see the working directory. Its `/var/www/html`.

**Reply by anupsathya** [reply]
Thanks! How does this translate to an externally accessible URL? When I do /var/www/html on certbot it says directory does not exist.

**Reply by JC5** [reply]
What you need for certbot is a folder to use and host the necessary `.well-known` directory. But a Docker container is just a port that you forward to, from a reverse proxy.

If you want to set up a Let's Encrypt certificate for Firefly III, you need to do so in the reverse proxy, not in Firefly III itself.

**Reply by anupsathya** [reply]
Ah, I see. Do you know of any guides that can help me with this when I'm using the standard docker-compose installation? I'm probably not using the right search terms but I can't seem to find any. Thanks a lot!

**Reply by JC5** [reply]
What are you using as a reverse proxy?

**Reply by anupsathya** [reply]
I'm not using anything at the time being really. I'm just pointing a subdomain to the server.

**Reply by JC5** [reply]
You shouldn't. If you want certbot to work, you need to `bash` into the container, install lets encrypt, configure it, configure Apache and restart the container to make it work. You'll have to do that again with every update. Or when you prune images and delete the Firefly III container.

If you use an reverse proxy, the lets encrypt config and everything is persistent, while Firefly III updates behind it. Stuff like nginx or Traefik isn't hard to set up.

**Reply by anupsathya** [reply]
Got it. I'll set it up. Thanks a lot!
